test/bar-horizontal.spec.js

```diff
@@ -0,0 +1,117 @@
+// --------------------------------------------------
+// IMPORT MODULES
+// --------------------------------------------------
+// Node
+const assert = require( 'assert' );
+
+// Vendor
+const figures = require( 'figures' );
+const capcon = require( 'capture-console' );
+
+// Project
+const barHorizontal = require( '../' );
+
+
+// --------------------------------------------------
+// DECLARE TESTS
+// --------------------------------------------------
+describe( 'barHorizontal', function() {
+  it( 'Should export a function.', function() {
+    assert.equal( typeof barHorizontal, 'function' );
+  } );
+
+
+  it( 'Should exit when invoked with 0x arguments.', function() {
+    assert.throws( function() {
+      barHorizontal();
+    } );
+  } );
+
+
+  it( 'Should exit when invoked with 1x invalid argument.', function() {
+    assert.throws( function() {
+      barHorizontal( 'This is an invalid argument.' );
+    } );
+  } );
+
+
+  it( 'Should exit when invoked with an empty array argument.', function() {
+    assert.throws( function() {
+      barHorizontal( [] );
+    } );
+  } );
+
+
+  it( 'Should exit when invoked with an empty object argument.', function() {
+    assert.throws( function() {
+      barHorizontal( {} );
+    } );
+  } );
+
+
+  it( 'Should run as normal when invoked with 1x non-empty array argument.', function() {
+    assert.doesNotThrow( function() {
+      barHorizontal( [ 1,2,3 ] );
+    } );
+  } );
+
+
+  it( 'Should run as normal when invoked with a 1x object argument', function() {
+    assert.doesNotThrow( function() {
+      barHorizontal( { a: 1, b: 2, c: 3 } );
+    } );
+  } );
+
+
+  it( 'Should accept an `options` object as an optional second argument.', function() {
+    assert.doesNotThrow( function() {
+      return barHorizontal( [ 1,2,3 ], {} );
+    } );
+  } );
+
+
+  it( 'Should not display labels by default.', function() {
+    let stdout = capcon.captureStdout( function() {
+      return barHorizontal( { a: 1, b: 2, c: 3 } );
+    } );
+
+    assert.equal( stdout.indexOf( 'a' ), -1 );
+  } );
+
+
+  it( 'Should display labels when invoked with the `labels: true` key/value pair.', function() {
+    let stdout = capcon.captureStdout( function() {
+      return barHorizontal( { a: 1, b: 2, c: 3 }, { labels: true } );
+    } );
+
+    assert.ok( stdout.indexOf( 'a' ) !== -1 );
+  } );
+
+
+  xit( 'Should not display warnings by default.', function() {
+    /// TODO: Unclear how to test for this, as warnings may not be triggered even when enabled.
```

Comment:
You can use the `width` option. For eg, `barHorizontal([100, 10], { width: 4, warnings: true })` will always raise a warning. (Please rebase onto latest code, I had goofed up the check for warnings in my last release)

Reply:
Awesome. Will revisit and update, thanks.